### ISSUE TITLE
When merging arrays containing objects the result is no longer an array but a object

### ISSUE THREAD

**Issue by koenpeters** [voodoocreation/ts-deepmerge#13]
How to replicate:
```
const a = [ { a: 1 } ]
const b = [ { a: 2 } ]
const c = merge.withOptions({ mergeArrays: false }, a, b)
console.log(c)
```

returns 
```
{ '0': { a: 2 } }
```

expected:
```
[ { a: 2 } ]
```

**Comment by voodoocreation** [comment]
Interesting - I wasn't aware that anyone was providing arrays as args - the implementation was supposed to only cater for objects and didn't realise that the args' interface wasn't safeguarding this until now.

Is there a reason you're not just spreading the two arrays like `const c = [...a, ...b];`? I can't see a deep merge being particularly helpful in this scenario, especially if `mergeArrays` is set to `false`. Let me know if there's anything I've missed/failed to consider though.

**Comment by voodoocreation** [comment]
Have just released `2.0.0` which fixes the arg interface to properly safeguard against arrays being provided as args. I can't think of a real world use case where this would be needed, but if you have a valid one, I can look into adding the support, though I imagine deep merging arrays would get very messy and confusing.

This helper was only designed to deep merge objects, which is why [the implementation doesn't dive into trying to deep merge arrays at all](https://github.com/voodoocreation/ts-deepmerge/blob/master/src/index.ts#L31-L33) - it just creates a new array with the values of the previous + current values (or just replaces it if `mergeArrays` is `false`). Hope that clears things up a bit though.
